ddl/partition.go

```diff
@@ -417,3 +417,16 @@ func isRangePartitionColUnsignedBigint(cols []*table.Column, pi *model.Partition
 	}
 	return false
 }
+
+// truncateTableFromReassignPartitionIDs reassign a new partition ids.
+func truncateTableFromReassignPartitionIDs(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo) error {
+	for i := range tblInfo.Partition.Definitions {
+		pid, err := t.GenGlobalID()
+		if err != nil {
+			job.State = model.JobStateCancelled
+			return errors.Trace(err)
+		}
+		tblInfo.Partition.Definitions[i].ID = pid
```

Comment:
So, if an error occur and DDL rollback, `model.TableInfo.Partition` is partially changed, some id is new and the old ?

Reply:
@tiancaiamao Only one `updateSchemaVersion` was executed in `onTruncateTable`, so if an error occurs and DDL rollback will not be a problem.